baseapp/abci.go

```diff
@@ -264,15 +278,15 @@ func (app *BaseApp) PrepareProposal(req abci.RequestPrepareProposal) (resp abci.
 		panic("PrepareProposal called with invalid height")
 	}
 
-	gasMeter := app.getBlockGasMeter(app.prepareProposalState.ctx)
-	ctx := app.getContextForProposal(app.prepareProposalState.ctx, req.Height)
-
-	ctx = ctx.WithVoteInfos(app.voteInfos).
+	app.prepareProposalState.ctx = app.getContextForProposal(app.prepareProposalState.ctx, req.Height).
```

Comment:
So now we set `app.prepareProposalState.ctx` in `PrepareProposal`. Is this intentional?